src/Core/Migration/V6_7/Migration1739355802FixVatHandling.php

```diff
@@ -0,0 +1,72 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Migration\V6_7;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Migration\MigrationStep;
+
+/**
+ * @internal
+ *
+ * @codeCoverageIgnore
+ */
+#[Package('framework')]
+class Migration1739355802FixVatHandling extends MigrationStep
+{
+    /**
+     * @var array<string, string>
+     */
+    private array $countryIds;
+
+    public function getCreationTimestamp(): int
+    {
+        return 1739355802;
+    }
+
+    public function update(Connection $connection): void
+    {
+        $this->addCountryVatPattern($connection);
+    }
+
+    private function addCountryVatPattern(Connection $connection): void
+    {
+        $this->fetchCountryIds($connection);
+
+        foreach ($this->getCountryVatPattern() as $isoCode => $countryVatPattern) {
+            if (!\array_key_exists($isoCode, $this->countryIds)) {
+                // country was deleted by shop owner
+                continue;
+            }
+
+            $connection->update('country', ['vat_id_pattern' => $countryVatPattern], ['id' => $this->countryIds[$isoCode]]);
+        }
+    }
+
+    private function fetchCountryIds(Connection $connection): void
+    {
+        /** @var list<array{id: string, iso: string}> $countries */
+        $countries = $connection->executeQuery('SELECT `id`, `iso` FROM `country`')->fetchAllAssociative();
+
+        foreach ($countries as $country) {
+            $this->countryIds[$country['iso']] = $country['id'];
+        }
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    private function getCountryVatPattern(): array
+    {
+        return [
+            'BE' => '(BE)?\d{10}',
+            'CY' => '(CY)?\d{8}[A-Z]',
+            'GR' => '(EL|GR)?\d{9}',
+            'HR' => '(HR)?\d{11}',
+            'IE' => '(IE)?(\d[0-9A-Z]\d{5}[A-W]|\d{7}[A-Z]{2})', // pre and post 2013 pattern
+            'LT' => '(LT)?(\d{9}|\d{12})',
+            'RO' => '(RO)?[1-9]\d{1,9}',
+            'XI' => '(XI)?(\d{9}(\d{3})?)', // Northern Ireland
```

Comment:
in no other VAT pattern, the country code is optional until now. Why start that?